fms_core/management/commands/curation.py

```diff
@@ -0,0 +1,93 @@
+        ACTION_ROLLBACK_CURATION: rollback_curation
+    }
+
+    def init_logging(self, log_name, timestamp):
+        filename = HOME + CURATION_PATH + LOG_PATH + timestamp + "_" + log_name + ".log"
+        formatter = logging.Formatter("%(asctime)s || %(levelname)s || %(message)s", datefmt="%Y-%m-%d %H:%M:%S")
+        handler = logging.FileHandler(filename, "w+")
+        handler.setFormatter(formatter)
+        log = logging.getLogger(log_name)
+        log.setLevel(logging.DEBUG)
+        log.addHandler(handler)
+        return log
+
+    def add_arguments(self, parser):
+        parser.add_argument("-p", dest="params_file", help="curation parameters file (json format)", required=True)
+
+    def handle(self, *args, **options):
+        self.stdout.write(self.style.SUCCESS("Started curation."))
+        # Copy the curation file to log directory with current timestamp matching log file
+        os.environ["TZ"] = "America/Montreal"
+        time.tzset()
+        timestamp = time.strftime("%Y-%m-%d_%H:%M:%S")
+        src_curation_file = HOME + CURATION_PATH + options["params_file"] + ".json"
+        dst_curation_file = HOME + CURATION_PATH + LOG_PATH + timestamp + "_" + options["params_file"] + ".json"
+        copyfile(src_curation_file, dst_curation_file)
+        # Use param file name to name log file
+        log = self.init_logging(options["params_file"], timestamp)
+        # log info identifying the curation
+        log.info("New curation started using identifier [" + options["params_file"] + "].")
+        # Extracting the curations to perform from file
+        with open(src_curation_file) as file:
+            params = json.load(file)
+        error_found = False
+        try:
+            with transaction.atomic():
+                # Create a curation revision to eventually rollback the current curation
+                with reversion.create_revision():
+                    # Launch each individual curation
+                    for curation in params:
+                        self.stdout.write(self.style.SUCCESS('Launching action [' + str(curation["curation_index"]) + '] "%s"' % curation["action"]) + '.')
+                        action = self.curation_switch.get(curation["action"])
+                        if action:
+                            curation_failed = action(curation, log)
+                            if curation_failed:
+                                self.stdout.write(self.style.ERROR("Action [" + str(curation_failed) + "] failed."))
+                                error_found = True
+                            else:
+                                self.stdout.write(self.style.SUCCESS("Action complete."))
+                        else:
+                            self.stdout.write(self.style.ERROR("Curation [" + str(curation["action"]) + "] do not exist."))
+                    reversion.set_user(User.objects.get(username="biobankadmin"))  # set biobankadmin (user 1) as user
```

Comment:
Move the admin username to settings.py as global var, don't hardcode it here.
Or something like it's done now in database connect settings https://github.com/c3g/freezeman_server/blob/master/fms/settings.py#L95

Reply:
A minor detail: I would suggest the same for other variables that were hardcoded, such as, for example "America/Montreal"